editoast/src/modelsv2/projects.rs

```diff
@@ -117,85 +117,86 @@ impl Project {
     use rstest::rstest;
-    use std::sync::Arc;
+    use std::ops::DerefMut;
 
     use super::*;
-    use crate::fixtures::tests::db_pool;
-    use crate::fixtures::tests::project;
-    use crate::fixtures::tests::TestFixture;
-    use crate::modelsv2::DbConnectionPool;
-    use crate::modelsv2::DeleteStatic;
+    use crate::modelsv2::fixtures::tests::create_test_project;
+    use crate::modelsv2::prelude::*;
+    use crate::modelsv2::DbConnectionPoolV2;
     use crate::modelsv2::Model;
-    use crate::modelsv2::Retrieve;
 
     #[rstest]
-    async fn create_delete_project(
-        #[future] project: TestFixture<Project>,
-        db_pool: Arc<DbConnectionPool>,
-    ) {
-        let project = project.await;
-        let conn = &mut db_pool.get().await.unwrap();
-        // Delete the project
-        assert!(Project::delete_static(conn, project.id()).await.unwrap());
-        // Second delete should be false
-        assert!(!Project::delete_static(conn, project.id()).await.unwrap());
+    async fn project_creation() {
+        let db_pool = DbConnectionPoolV2::for_tests();
+        let project_name = "test_project_name";
+        let created_project = create_test_project(db_pool.get_ok().deref_mut(), project_name).await;
+        assert_eq!(created_project.name, project_name);
     }
 
     #[rstest]
-    async fn get_project(#[future] project: TestFixture<Project>, db_pool: Arc<DbConnectionPool>) {
-        let fixture_project = &project.await.model;
-        let conn = &mut db_pool.get().await.unwrap();
+    async fn project_retrieve() {
+        let db_pool = DbConnectionPoolV2::for_tests();
+        let created_project =
+            create_test_project(db_pool.get_ok().deref_mut(), "test_project_name").await;
 
         // Get a project
-        let project = Project::retrieve(conn, fixture_project.id)
+        let project = Project::retrieve(db_pool.get_ok().deref_mut(), created_project.id)
             .await
-            .unwrap()
-            .unwrap();
-        assert_eq!(fixture_project, &project);
-        assert!(Project::list(conn, SelectionSettings::new()).await.is_ok());
+            .expect("Failed to retrieve project")
+            .expect("Project not found");
+
+        assert_eq!(&created_project, &project);
     }
 
     #[rstest]
-    async fn sort_project(#[future] project: TestFixture<Project>, db_pool: Arc<DbConnectionPool>) {
-        let project = project.await;
-        let project_2 = project
-            .model
+    async fn project_update() {
+        let db_pool = DbConnectionPoolV2::for_tests();
+        let created_project =
+            create_test_project(db_pool.get_ok().deref_mut(), "test_project_name").await;
+
+        let project_name = "update_name";
+        let project_budget = Some(1000);
+
+        // Patch a project
+        created_project
             .clone()
-            .into_changeset()
-            .name(project.model.name.clone() + "_bis");
+            .patch()
```

Comment:
you don't need to clone() it (am I missing something? 🤔)

Reply:
No, it wasn't necessary to clone. 👍